lib/src/local_working_copy.rs

```diff
@@ -1175,37 +1175,60 @@ impl TreeState {
     }
 
     #[cfg(feature = "watchman")]
-    #[tokio::main(flavor = "current_thread")]
     #[instrument(skip(self))]
     pub async fn query_watchman(
         &self,
         config: &WatchmanConfig,
     ) -> Result<(watchman::Clock, Option<Vec<PathBuf>>), TreeStateError> {
-        let fsmonitor = watchman::Fsmonitor::init(&self.working_copy_path, config)
-            .await
-            .map_err(|err| TreeStateError::Fsmonitor(Box::new(err)))?;
+        let working_copy_path = self.working_copy_path.clone();
```

Comment:
nit: perhaps, `.clone()` wouldn't be needed since the async code isn't spawned.

Reply:
you're right, this was just force of habit